reveal {map,mod,log} folder

themasch opened the pull request:
This would fix #514, #513 and #512. 

I'm not entirely sure about some aspects of this PR so this is more a **WIP**: 

 - This PR (hopefully) solves three (very similar) open issues and not only one. Should I split it? 
 - I added a static method `openFolder(Path)` to `com.faforever.client.io.FileUtils` which opens the windows explorer on a given path. This currently uses the `java.lang.ProcessBuilder` to call "explorer.exe <PATH>". Should that method be in that class? Should I add support for other OS? 
   - I tried using `java.awt.Desktop` to solve this but (a) using awt stuff in a swing/jfx app seems wrong and (b) I didn't work for me ("not supported").
- Instead of opening "the client's log file", it currently opens the FAF log folder. I'm not sure if thats *correct* and, if not, where to get the correct path. 
- Error handling: currently, there is none. Add a logger to the MainController and let it print the stacktraces?
- Testing: I'm not really sure on how to test this automatically.

themasch commented:
Okay, that travis failure is odd. Don't have it locally and it seems to fail in `testLoadInstalledModsDoesntUnloadsMods` which is sounds like its far away from everything I might have touched.

micheljung commented on src/main/java/com/faforever/client/io/FileUtils.java:
```diff
@@ -34,4 +34,11 @@ public FileVisitResult postVisitDirectory(Path dir, IOException e) throws IOExce
+    processBuilder.directory(path.toFile());
+    processBuilder.command("explorer.exe", path.toAbsolutePath().toString());
+    processBuilder.start();
+  }
```
You can remove this and use `PlatformService#reveal()` instead :)

micheljung commented on src/main/java/com/faforever/client/main/MainController.java:
```diff
@@ -456,4 +459,31 @@ public void onNavigateEvent(NavigateEvent navigateEvent) {
+      FileUtils.openFolder(mapPath);
+    } catch (IOException e) {
+      e.printStackTrace();
+    }
```
Don't do `e.printStackTrace()`, [as Codacy reports](https://www.codacy.com/app/FAForever/downlords-faf-client/pullRequest?prid=629971). But once you use the platform service there won't be an `IOException` anyway.

themasch commented:
Thanks for the helpful review. I updated the PR to use the PlatformService and added the tests. 
Sadly, I can't test it until this evening since the bridj lib doesn't know how to open files on todays linux systems. Could use an update to utilize `xdg-open` ;)

micheljung commented:
Very nice, thanks a lot! Keep the PRs coming :)
